sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala

```diff
@@ -368,6 +368,238 @@ class SqlScriptingInterpreterSuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  test("searched case") {
+    val commands =
+      """
+        |BEGIN
+        | CASE
+        |   WHEN 1 = 1 THEN
+        |     SELECT 42;
+        | END CASE;
+        |END
+        |""".stripMargin
+    val expected = Seq(Seq(Row(42)))
+    verifySqlScriptResult(commands, expected)
+  }
+
+  test("searched case nested") {
+    val commands =
+      """
+        |BEGIN
+        | CASE
+        |   WHEN 1=1 THEN
+        |   CASE
+        |    WHEN 2=1 THEN
+        |     SELECT 41;
+        |   ELSE
+        |     SELECT 42;
+        |   END CASE;
+        | END CASE;
+        |END
+        |""".stripMargin
+    val expected = Seq(Seq(Row(42)))
+    verifySqlScriptResult(commands, expected)
+  }
+
+  test("searched case second case") {
+    val commands =
+      """
+        |BEGIN
+        | CASE
+        |   WHEN 1 = (SELECT 2) THEN
+        |     SELECT 1;
+        |   WHEN 2 = 2 THEN
+        |     SELECT 42;
+        |   WHEN (SELECT * FROM t) THEN
```

Comment:
In this case, table `t` actually does not exist. Do we need to check if `this logical plan` can be `resolved`?

Reply:
In this case the table t not existing should not matter, because the case statement will never reach it, as it will (should) stop at 2 = 2.

Reply:
Yes, I'm actually curious. In SQL scripting, before we execute the script, should we `check` if the `logical plan `generated by the statement can be successfully `resolved`?

Reply:
I'm not quite sure about this.

Reply:
If you look at other sql scripting structures, like `while` and `if else`, they work the same way. This is because in the current architecture, the interpreting phase happens before the analyzer phase.

Reply:
@panbingkun to clarify a bit more, scripting works as following:
- parsing the text and creating a `CompoundBody` logical plan at the root
- in leaf statements (`SingleStatement`) logical plans of each separate SQL statement is saved
- interpreter iterates through the statements
- each leaf statement is executed based on the saved logical plan

it's really hard to do what you are asking, since `QueryExecution` (and thus analyzer, optimizer, etc) are created and designed to work on the level of a single query.
it would be too much to do what you suggested, but maybe sometime in the future if we figure out there's a need and what's the proper way to do so...